### ISSUE TITLE
Setting a single member of a class does not 'save'

### ISSUE THREAD

**Issue by RobDonovan** [krakjoe/pthreads#517]
Hi,

I posted this on SO, but I don't seem to have had any response, so I thought I would try here.

I'm not sure if its a bug with pthreads, or just I'm using it incorrectly.

If I try to set a single member of a class that I have defined as threaded, then it does not seem to save the data back to it.

I have a full explanation here: http://stackoverflow.com/questions/33519512/php-pthreads-and-using-a-class-to-store-data-that-is-shared-between-threads

Any help would be appreciated, thanks.

I also have another request (bare in mind, I'm new to pthreads, so maybe this is possible some way already)

Is it possible to have an enhancement that allows us to use synchronized at a lower level.

From my above example (commented at the bottom of the code), I have to used synchronized on $this->cacheData to enforce integrity, and I think this is basically creating a 'lock' on the whole cacheData dataset.

What I would like to do is just lock/synchronized  a certain 'record' from this 'array', so that my threads dont lock each other when they are accessing different records.

So, for example: $this->cacheData['record1']->synchronized(....);

Basically row locking instead of table locking.

This would then mean that the threads are not locking and waiting for each other so much, even when not trying to change the same 'record'. I see that it might not be usable for everyone, but I would certainly benefit from being able to do this.

**Comment by krakjoe** [comment]
member/row level locking not possible, or not possible in such a way that you would see any gains whatever.

If you poke around in examples, you can find the answers to your other questions ... when I have time, I'll do a stackoverflow answer ...

Closing the issue here, this is not the place to ask questions ...

**Comment by RobDonovan** [comment]
Ok, I see.

Well, I can use my own objects to sync with to 'lock' at a record level, it helps with performance in my case, just it would have been better if I didnt need to do it that way.

I've tried looking through the examples, but couldnt find anything to help me with my SO question, so when you have the time, it would be good to get a suggestion.

Thanks.
